src/main/java/org/sciborgs1155/lib/InputStream.java

```diff
@@ -0,0 +1,124 @@
+package org.sciborgs1155.lib;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import java.util.function.DoubleSupplier;
+import java.util.function.DoubleUnaryOperator;
+
+/**
+ * A functional interface to aid in modifying double suppliers, such as from a joystick.
+ *
+ * <p>Inspired by 694's StuyLib.
+ */
+@FunctionalInterface
+public interface InputStream extends DoubleSupplier {
+
+  /**
+   * Creates an input stream from another.
+   *
+   * @param base The base stream.
+   * @return A new input stream.
+   */
+  public static InputStream of(InputStream base) {
+    return base;
+  }
+
+  /**
+   * Shorthand to return a double value.
+   *
+   * @return The value from {@link #getAsDouble()}.
+   */
+  public default double get() {
+    return getAsDouble();
+  }
+
+  /**
+   * Transforms the stream outputs by an operator.
+   *
+   * @param operator A function that takes in two double inputs and returns one.
```

Comment:
pretty sure this is a typo not a conceptual error, but a DoubleUnaryOperator takes one double input, not two